code/procs/jobprocs.dm

```diff
@@ -396,6 +409,34 @@
 				if(SL.len > 0)
 					src.set_loc(pick(SL))
 
+			if (src.traitHolder && src.traitHolder.hasTrait("pilot"))		//Has the Pilot trait - they're drifting off-station in a pod. Note that environmental checks are not needed here.
+				var/turf/pilotSpawnLocation = null
+				
+				#ifdef UNDERWATER_MAP										//This part of the code executes only if the map is a water one.
+				while(!pilotSpawnVerif(pilotSpawnLocation))					//Trying to find a valid spawn location.
+					pilotSpawnLocation = locate(rand(1, world.maxx), rand(1, world.maxy), 5)
+				if (pilotSpawnLocation)										//Sanity check.
+					src.set_loc(pilotSpawnLocation)
+				var/obj/machinery/vehicle/tank/minisub/V = new/obj/machinery/vehicle/tank/minisub/pilot(pilotSpawnLocation)
+				#else														//This part of the code executes only if the map is a space one.
+				var/pilotZ = rand(2, (world.maxz - 1))
```

Comment:
```suggestion
				var/pilotZ = rand(2, world.maxz)
```
If you insist on keeping this you should at least cover all relevant z-levels.

Reply:
The Z-level that is world.maxz is always one with the football field, and it would be immersion-breaking to spawn there, hence -1.

Reply:
After a chat, we've landed on the decision to define every single Z-level and pick from the ones we need.